File: 3189-find-champion-ii/find-champion-ii.py
class Solution:
    def findChampion(self, n: int, edges: list[list[int]]) -> int:
        count = 0
        dp = [0] * n

        for edge in edges:
            dp[edge[1]] += 1

        champion = -1

        for i in range(n):
            if dp[i] == 0:
                count += 1
                champion = i

        return champion if count == 1 else -1

File: 3189-find-champion-ii/find-champion-ii.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @return {number}
 */
var findChampion = function (n, edges) {
    let dp = new Array(n).fill(0);
    let count = 0, champion = -1;

    for (const edge of edges) {
        dp[edge[1]] += 1;
    }

    for (let i = 0; i < n; i++) {
        if (dp[i] === 0) {
            count += 1;
            champion = i;
        }
    }

    return count === 1 ? champion : -1;
};